debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java

```diff
@@ -218,7 +226,7 @@ public ChainedLogicalStreamBuilder optionsWithoutMetadata(ChainedLogicalStreamBu
     private void handleBeginMessage(ByteBuffer buffer, ReplicationMessageProcessor processor) throws SQLException, InterruptedException {
         final Lsn lsn = Lsn.valueOf(buffer.getLong()); // LSN
         this.commitTimestamp = PG_EPOCH.plus(buffer.getLong(), ChronoUnit.MICROS);
-        this.transactionId = buffer.getInt();
+        this.transactionId = (long) buffer.getInt();
```

Comment:
@jpechane, wondering whether we actually did have an issue here; IIRC, tx ids are unsigned int32. We may have exported values larger than Integer.MAX_VALUE as negative values before?

Reply:
Ok, pushed a commit for converting to unsigned long.